pkg/sm/client.go

```diff
@@ -43,7 +44,7 @@ var ErrBrokerPlatformCredentialsNotFound = errors.New("credentials not found")
 //go:generate counterfeiter . Client
 type Client interface {
 	GetBrokers(ctx context.Context) ([]*types.ServiceBroker, error)
-	GetVisibilities(ctx context.Context) ([]*types.Visibility, error)
+	GetVisibilities(ctx context.Context, planIDs []string) ([]*types.Visibility, error)
 	GetPlans(ctx context.Context) ([]*types.ServicePlan, error)
```

Comment:
maybe leave old GetVisibilities for agents that need all plans (r.options.VisibilityBrokerChunkSize == 0)because query with many parameters in IN clause can be performance killer
https://stackoverflow.com/questions/1013797/is-sql-in-bad-for-performance

Reply:
I think you are right. I must change it to be the prev imple in case VisibilityBrokerChunkSize == 0. besides the potential performance issue you raised, I think the query might fail on "too long" query

Reply:
done